ReplayKitExample/ReplayKitExample/ReplayKitVideoSource.swift

```diff
@@ -63,74 +64,64 @@ class ReplayKitVideoSource: NSObject, TVIVideoCapturer {
 
     init(isScreencast: Bool) {
         screencastUsage = isScreencast
-    }
 
-    public var isScreencast: Bool {
-        get {
-            return screencastUsage
-        }
-    }
+        let outputFormat = TVIVideoFormat()
 
-    public var supportedFormats: [TVIVideoFormat] {
-        get {
-            /*
-             * Describe the supported formats.
-             * In this example we can deliver either original, or downscaled buffers.
-             */
+        if (!isScreencast) {
+            outputFormat.frameRate = UInt(ReplayKitVideoSource.kFormatFrameRate)
+            outputFormat.pixelFormat = TVIPixelFormat.formatYUV420BiPlanarFullRange
+            outputFormat.dimensions = CMVideoDimensions(width: Int32(ReplayKitVideoSource.kDownScaledMaxWidthOrHeight),
+                                                       height: Int32(ReplayKitVideoSource.kDownScaledMaxWidthOrHeight))
+        } else {
             var screenSize = UIScreen.main.bounds.size
             screenSize.width *= UIScreen.main.nativeScale
             screenSize.height *= UIScreen.main.nativeScale
-            let format = TVIVideoFormat()
-            format.pixelFormat = TVIPixelFormat.formatYUV420BiPlanarFullRange
-            format.frameRate = UInt(ReplayKitVideoSource.kFormatFrameRate)
-            format.dimensions = CMVideoDimensions(width: Int32(screenSize.width), height: Int32(screenSize.height))
-
-            // We will downscale buffers to a 640x640 box, if requested.
-            let downscaledFormat = TVIVideoFormat()
-            downscaledFormat.frameRate = UInt(ReplayKitVideoSource.kFormatFrameRate)
-            downscaledFormat.pixelFormat = TVIPixelFormat.formatYUV420BiPlanarFullRange
-            downscaledFormat.dimensions = CMVideoDimensions(width: Int32(ReplayKitVideoSource.kDownScaledMaxWidthOrHeight), height: Int32(ReplayKitVideoSource.kDownScaledMaxWidthOrHeight))
-
-            return [format, downscaledFormat]
+            outputFormat.pixelFormat = TVIPixelFormat.formatYUV420BiPlanarFullRange
+            outputFormat.frameRate = UInt(ReplayKitVideoSource.kFormatFrameRate)
+            outputFormat.dimensions = CMVideoDimensions(width: Int32(screenSize.width), height: Int32(screenSize.height))
         }
+
+        super.init()
+        requestOutputFormat(outputFormat)
```

Comment:
In this case, I wouldn't request an output format unless downscaling is the user's intention.